app/lib/features/bluetooth/backend/bluetooth_connection.dart

```diff
@@ -0,0 +1,26 @@
+part of 'bluetooth_backend.dart';
+
+/// State of the bluetooth connection of a device
+enum BluetoothConnectionState {
+  /// Device is connected
+  connected,
+  /// Device is disconnect
+  disconnected;
+}
+
+/// Bluetooth connection state parser base class
+///
+/// This is a separate helper class as factory or static methods cannot be abstract,
+/// so even though this class only has one method it's useful to enforce the types
+abstract class BluetoothConnectionStateParser<BackendConnectionState> extends StreamDataParser<BackendConnectionState, BluetoothConnectionState> {}
+
+/// Transforms the backend's bluetooth adapter state stream to emit [BluetoothAdapterState]'s
+///
+/// Can normally be used directly, backends should only inject a customized [BluetoothAdapterStateParser]
+class BluetoothConnectionStateStreamTransformer<BackendConnectionState, BASP extends BluetoothConnectionStateParser<BackendConnectionState>>
```

Comment:
> btw, moved everything to map's

No you didn't 🙃

> afaik transform is 'more' correct so should be fine to keep it

It really makes reading the code more difficult. 

The old tests could be deleted and we can rely on implicit testing, if that's what you're concerned about. This shouldn't slow down debugging too much and might even make it faster because there are less jumps when reading the code.